### DISCUSSION TITLE
Remove battery related properties from the vacuum entity

### DISCUSSION THREAD

**Discussion by emontnemery** [home-assistant/architecture#938] [Entity Models] [answered]
### Suggestion:
Remove battery related properties `battery_icon` and `battery_level` from the vacuum entity.

### Rationale:
Instead of exposing the vacuum cleaner's battery status as vacuum entity state attributes, the battery status should be exposed as a separate `sensor` entity with its device class set to battery

**Comment by frenck** [comment, marked as answer]
Matches all the things we have been doing with batteries already. 👍 

../Frenck

**Comment by piitaya** [comment]
Does it need a `binary_sensor` with `charging` device class to known when the vacuum is charging? We had this information with the `battery_icon` before and I think it will be cleaner with a dedicated charging binary sensor.

**Comment by jherby2k** [comment]
New suggested repair:

> Vacuum entity vacuum.valetudo_xxxx implements the battery feature which is deprecated. This will stop working in Home Assistant 2026.2. Implement a separate entity for the battery state instead. To fix the issue, remove the battery feature from the configured supported features, and restart Home Assistant.

I can't figure out how or where to change the "configured supported features". It doesn't appear to be in the MQTT integration options.

**Reply by rytilahti** [reply]
As far as I know, you have to adjust it at the source, i.e., in valetudo.
